identity-iota/src/tangle/client.rs

```diff
@@ -74,23 +76,37 @@ impl Client {
   }
 
   /// Publishes an [`IotaDocument`] to the Tangle.
-  pub async fn publish_document(&self, document: &IotaDocument) -> Result<Receipt> {
-    self.publish_json(document.integration_index(), document).await
+  pub async fn publish_document(&self, document: &IotaDocument, compression: bool) -> Result<Receipt> {
+    self
+      .publish_json(document.integration_index(), document, compression)
+      .await
   }
 
   /// Publishes a [`DocumentDiff`] to the Tangle to form part of the diff chain for the integration
   /// chain message specified by the given [`MessageId`].
-  pub async fn publish_diff(&self, message_id: &MessageId, diff: &DocumentDiff) -> Result<Receipt> {
-    self.publish_json(&IotaDocument::diff_index(message_id)?, diff).await
+  pub async fn publish_diff(&self, message_id: &MessageId, diff: &DocumentDiff, compression: bool) -> Result<Receipt> {
+    self
+      .publish_json(&IotaDocument::diff_index(message_id)?, diff, compression)
+      .await
   }
 
-  /// Publishes arbitrary JSON data to the specified index on the Tangle.
-  pub async fn publish_json<T: ToJson>(&self, index: &str, data: &T) -> Result<Receipt> {
+  /// Compresses and Publishes arbitrary JSON data to the specified index on the Tangle.
```

Comment:
I think `compression` should be changed to be on the `Client` instead of the `ClientMap`. None of these methods would need a `compression` parameter then and `publish_json` would be the single place it is accessed.

Reply:
The way the clients are stored inside the `ClientMap` makes them hard to mutate. Storing them in a `Mutex` for example looked more complicated.

Reply:
Fair point.

Reply:
Now, disabling compression can only be done through the `Client` directly. And not through the `ClientMap`. But a `Client` can be added to a `ClientMap` manually. For the WASM bindings, it can be set through `Config`.